type.go

```diff
@@ -45,6 +45,9 @@ type resultMaker func(reflect.Value) reflect.Value
 // but passed to makeResult and then updated.
 type unmarshaler func(v reflect.Value, p Params, makeResult resultMaker) error
 
+// marshaler marshals the specified value into params.
```

Comment:
After my remarks above, I would suggest this:

    // The value is always the value type, even if the field type
    // is a pointer.